tests/nlp/checks/data_integrity/frequent_substrings_test.py

```diff
@@ -0,0 +1,147 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Test for the NLP FrequentSubstrings check."""
+
+import pytest
+from hamcrest import *
+
+from deepchecks.nlp.checks.data_integrity.frequent_substrings import FrequentSubstrings
+from deepchecks.nlp.text_data import TextData
+from tests.base.utils import equal_condition_result
+
+
+@pytest.fixture
+def dataset_without_frequent_substrings():
+    return TextData(
+        raw_text=[
+            'Hello world',
+            'Do not worry be happy',
+            'Explicit is better than implicit'
+        ]
+    )
+
+
+@pytest.fixture
+def dataset_with_equal_frequent_substrings():
+    return TextData(
+        raw_text=[
+            'Explicit is better than implicit.',
+            'Do not worry be happy',
+            'Explicit is better than implicit...',
+            'Simple is better than complex',
+            'Simple is better than',
+            'He said - Do not worry be happy'
+        ]
+    )
+
+
+@pytest.fixture
+def dataset_with_similar_frequent_substrings():
+    return TextData(
+        raw_text=[
+            'Do not worry be happy. Explicit is better than implicit.',
+            'Explicit is better than implicit...',
+            'Simple is better than complex',
+            'Simple is better than- Do not worry be happy',
+            'Explicit is better than implicit.',
+            'Do not worry be happy',
+            'Explicit is better than implicit...',
+            'Simple is better than complex',
+            'Do not worry be happy!',
+            'He said - Do not worry be happy'
+        ]
+    )
+
+
+@pytest.fixture
+def dataset_with_empty_string():
+    return TextData(
+        raw_text=[
+            ''
+        ]
+    )
+
+# =================
+# ----- Tests -----
+# =================
```

Comment:
Additional comment - add 1-2 tests about the return value of the check (currently all the tests check for the result of the condition)